src/core/src/op/cum_sum.cpp

```diff
@@ -58,8 +57,65 @@ shared_ptr<Node> op::v0::CumSum::clone_with_new_inputs(const OutputVector& new_a
     }
 }
 
-NGRAPH_SUPPRESS_DEPRECATED_START
+OPENVINO_SUPPRESS_DEPRECATED_START
 shared_ptr<Node> op::v0::CumSum::get_default_value() const {
     return ngraph::make_constant_from_string("0", get_element_type(), get_shape());
 }
-NGRAPH_SUPPRESS_DEPRECATED_END
+OPENVINO_SUPPRESS_DEPRECATED_END
+
+namespace {
+template <element::Type_t DATA_ET, element::Type_t AXIS_ET>
+bool evaluate_cum_sum(TensorVector& outputs, const TensorVector& inputs, const bool exclusive, const bool reverse) {
+    using data_t = fundamental_type_for<DATA_ET>;
+    using axis_t = fundamental_type_for<AXIS_ET>;
+    ngraph::runtime::reference::cumsum<data_t, axis_t>(inputs[0].data<data_t>(),
+                                                       inputs[1].data<axis_t>(),
+                                                       outputs[0].data<data_t>(),
+                                                       inputs[0].get_shape(),
+                                                       exclusive,
+                                                       reverse);
+    return true;
+}
+
+#define CUM_SUM_TYPE_CASE(a, ...)                                          \
+    case element::Type_t::a: {                                             \
+        OV_OP_SCOPE(OV_PP_CAT4(evaluate_cum_sum, _, a, AXIS_ET));          \
+        return evaluate_cum_sum<element::Type_t::a, AXIS_ET>(__VA_ARGS__); \
+    }
+
+template <element::Type_t AXIS_ET>
+bool evaluate(TensorVector& outputs, const TensorVector& inputs, const bool exclusive, const bool reverse) {
+    switch (inputs[0].get_element_type()) {
+        CUM_SUM_TYPE_CASE(f32, outputs, inputs, exclusive, reverse);
+    default:
+        return false;
+    }
+}
+
+bool evaluate_cum_sum(TensorVector& outputs, const TensorVector& inputs, const bool exclusive, const bool reverse) {
+    auto rc = true;
+    switch (inputs[1].get_element_type()) {
+        NGRAPH_TYPE_CASE(evaluate_cum_sum, i32, outputs, inputs, exclusive, reverse);
+        NGRAPH_TYPE_CASE(evaluate_cum_sum, i64, outputs, inputs, exclusive, reverse);
+    default:
+        return false;
+    }
+    return rc;
+}
+}  // namespace
+
+bool op::v0::CumSum::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+    OV_OP_SCOPE(v0_CumSum_evaluate);
+    OPENVINO_ASSERT(inputs.size() == 2, "Invalid CumSum inputs TensorVector.");
```

Comment:
nitpick: Invalid size of  `inputs` argument of `evaluate` method of CumSum operation. Provided size: ... . Expected: .

Reply:
updated